akka-docs/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -186,33 +187,44 @@ class HttpServerExampleSpec extends WordSpec with Matchers {
     import akka.http.scaladsl.model.HttpMethods._
     import akka.http.scaladsl.model._
     import akka.stream.ActorMaterializer
+    import scala.io.StdIn
 
-    implicit val system = ActorSystem()
-    implicit val materializer = ActorMaterializer()
+    object WebServer {
 
-    val requestHandler: HttpRequest => HttpResponse = {
-      case HttpRequest(GET, Uri.Path("/"), _, _, _) =>
-        HttpResponse(entity = HttpEntity(ContentTypes.`text/html(UTF-8)`,
-          "<html><body>Hello world!</body></html>"))
+      def main(args: Array[String]) {
+        implicit val system = ActorSystem()
+        implicit val materializer = ActorMaterializer()
+        implicit val ec = system.dispatcher
 
-      case HttpRequest(GET, Uri.Path("/ping"), _, _, _) =>
-        HttpResponse(entity = "PONG!")
+        val requestHandler: HttpRequest => HttpResponse = {
+          case HttpRequest(GET, Uri.Path("/"), _, _, _) =>
+            HttpResponse(entity = HttpEntity(ContentTypes.`text/html(UTF-8)`,
+              "<html><body>Hello world!</body></html>"))
 
-      case HttpRequest(GET, Uri.Path("/crash"), _, _, _) =>
-        sys.error("BOOM!")
+          case HttpRequest(GET, Uri.Path("/ping"), _, _, _) =>
+            HttpResponse(entity = "PONG!")
 
-      case _: HttpRequest =>
-        HttpResponse(404, entity = "Unknown resource!")
-    }
+          case HttpRequest(GET, Uri.Path("/crash"), _, _, _) =>
+            sys.error("BOOM!")
+
+          case _: HttpRequest =>
+            HttpResponse(404, entity = "Unknown resource!")
+        }
+
+        val bindingFuture = Http().bindAndHandleSync(requestHandler, "localhost", 8080)
+        StdIn.readLine() // let it run until user presses return
```

Comment:
Also, It's not quite true to do this in the line after, should be in a `Future.foreach` or else we don't know it actually did start. But that makes the sample noisier so I'm leaving it be as is with just printlns.